### DISCUSSION TITLE
Should Silk.NET.Maths be used instead of System.Numerics for consistency?

### DISCUSSION THREAD

**Discussion by qian-o** [dotnet/Silk.NET#1602] [General] [answered]
Hello, I have noticed a mix of usage in the code, where Maths and Numerics are being used together. For example, float types are using Numerics and int types are using Maths.

Should this be unified? It may make the code cleaner for users.

**Comment by Beyley** [comment, marked as answer]
avoid Silk.NET.Maths, if you *can* avoid it, do so, System.Numerics is faster and will always be faster than non-SIMD Silk.NET.Maths
it exists to extend missing functionality, not replace existing functionality provided by System.Numerics

**Reply by qian-o** [reply]
Thank you for answering my questions. I think I know how to weigh the options between these two libraries now.

**Reply by BuslikDrev** [reply]
Есть пример редактирования каждого пикселя изображения?
Например GD+
System.Numerics.Matrix4x4
System.Numerics.Vector3.Transform
System.Runtime.InteropServices.Marshal
System.Collections.Generic.IEnumerator<int[]>
e.Graphics.DrawImage
System.Drawing.Color

Изображение 1024x1024 (массив из 1048576 пикселей) обрабатывается за 30 мс при работе на 4 ядрах процессора или за 105 мс на одном ядре.
Потребление оперативной памяти от 80-180 Мб.

![image](https://github.com/dotnet/Silk.NET/assets/32706609/a282d0c4-8417-4472-bbba-30099b645516)
